web3_proxy/src/rpcs/request.rs

```diff
@@ -194,15 +195,35 @@ impl OpenRequestHandle {
 
         // TODO: replace ethers-rs providers with our own that supports streaming the responses
         // TODO: replace ethers-rs providers with our own that handles "id" being null
-        let response: Result<R, _> = if let Some(ref p) = self.rpc.http_provider {
-            p.request(method, params).await
-        } else if let Some(p) = self.rpc.ws_provider.load().as_ref() {
-            p.request(method, params).await
-        } else {
-            return Err(ProviderError::CustomError(
-                "no provider configured!".to_string(),
-            ));
-        };
+        let response: Result<StreamingResponse<R>, _> =
+            if let Some(ref client) = self.rpc.http_client {
+                let id = self
+                    .rpc
+                    .next_id
+                    .fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+                let payload = JsonRpcRequest::new(
+                    JsonRpcId::Number(id),
+                    method.to_string(),
+                    serde_json::to_value(params)?,
+                )
+                .unwrap();
+                // TODO no early return
+                let resp = client
+                    .post(self.rpc.http_url.clone().expect("must have url"))
```

Comment:
can we prepare the client with this url already included in it? seems like we shouldn't need this string every time